lib/components/Input.js

```diff
@@ -1,121 +1,127 @@
-import React, { Component } from "react";
+import React from "react";
 import classnames from "classnames";
 import Label from "./Label";
 
-class Input extends Component {
-  render() {
-    const {
-      size,
-      type,
-      label,
-      value,
-      error,
-      suffix,
-      prefix,
-      disabled,
-      helpText,
-      onChange,
-      className,
-      inputClassName,
-      naked,
-      maxLength,
-      labelProps,
-      dataTestId,
-      placeholder,
-      leadingIcon,
-      required = false,
-      hideCharacterCount,
-      ...otherProps
-    } = this.props;
+const Input = React.forwardRef((props, ref) => {
+  const {
+    size,
+    type,
+    label,
+    value,
+    error,
+    suffix,
+    prefix,
+    disabled,
+    helpText,
+    onChange,
+    className,
+    inputClassName,
+    maxLength,
+    labelProps,
+    nakedInput,
+    dataTestId,
```

Comment:
We should be removing this prop from all our components. We should be passing it as `data-test-id` itself and pass it along as `otherProps`

Reply:
@goutham-subramanyam 

Will handle this is in a separate issue if it needs to be removed from all components. Or are you suggesting to make that change in this PR itself? 

If in this PR itself, then there will be a lot of changes across many components.

Awaiting your response.

Reply:
You can remove dataTestID for Input in this PR and handle the rest in separate issue.
@yedhink _a